wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -0,0 +1,487 @@
+
+    message.fields().forEach { field ->
+      var fieldClass: TypeName = typeName(field.type())!!
+      val fieldName = nameAllocator.get(field)
+      var defaultValue = CodeBlock.of("null")
+
+      when {
+        field.isOptional -> {
+          fieldClass = fieldClass.asNullable()
+        }
+        field.isRepeated -> {
+          fieldClass = List::class.asClassName().parameterizedBy(fieldClass)
+          defaultValue = CodeBlock.of("emptyList()")
+        }
+      }
+
+      if (field.default != null) {
+        defaultValue = getDefaultValue(field)
+        fieldClass = fieldClass.asNonNullable()
+      }
+
+      val parameterSpec = ParameterSpec.builder(fieldName, fieldClass)
+      if (field.isOptional || field.isRepeated) {
+        parameterSpec.defaultValue(defaultValue)
+      }
+
+      constructorBuilder.addParameter(parameterSpec.build())
+      classBuilder.addProperty(PropertySpec.builder(fieldName, fieldClass)
+          .initializer(fieldName)
+          .build())
+    }
+
+    val unknownFields = nameAllocator.get("unknownFields")
+    constructorBuilder.addParameter(
+        ParameterSpec.builder(unknownFields, byteClass)
+            .defaultValue("%T.EMPTY", byteClass)
+            .build())
+    classBuilder.addProperty(PropertySpec.builder(unknownFields, byteClass)
+        .initializer(unknownFields)
+        .build())
+
+    classBuilder.primaryConstructor(constructorBuilder.build())
+  }
+
+
+  /**
+   * Example
+   * object ADAPTER : ProtoAdapter<Person>(FieldEncoding.LENGTH_DELIMITED, Person::class.java) {
+   *     override fun encodedSize(value: Person): Int { .. }
+   *     override fun encode(writer: ProtoWriter, value: Person) { .. }
+   *     override fun decode(reader: ProtoReader): Person { .. }
+   * }
+   */
+  private fun generateAdapter(type: MessageType): TypeSpec {
+    val nameAllocator = nameAllocator(type)
+    val parentClassName = generatedTypeName(type)
+    val adapterName = nameAllocator.get("ADAPTER")
+
+    return TypeSpec.objectBuilder(adapterName)
+        .superclass(ProtoAdapter::class.asClassName().parameterizedBy(parentClassName))
+        .addSuperclassConstructorParameter("%T.%L", FieldEncoding::class.asClassName(),
+            "LENGTH_DELIMITED")
+        .addSuperclassConstructorParameter("%T::class.java", parentClassName)
+        .addFunction(encodedSizeFunc(type))
+        .addFunction(encodeFunc(type))
+        .addFunction(decodeFunc(type))
+        .build()
+  }
+
+  private fun encodedSizeFunc(message: MessageType): FunSpec {
+    val className = generatedTypeName(message)
+    val body = CodeBlock.builder()
+        .add("return ")
+
+    val indentation = " ".repeat(4)
+
+    message.fields().forEach { field ->
+      val adapterName = getAdapterName(field)
+
+      body.add("%L.%LencodedSizeWithTag(%L, value.%L) +\n",
+          adapterName, if (field.isRepeated) "asRepeated()." else "", field.tag(), field.name())
+      body.add("%L", indentation)
```

Comment:
I think you just want to add `%[` and `%]` in the line, which causes wraps to be indented more